ConsoleGUI/Controls/VerticalScrollPanel.cs

```diff
@@ -105,19 +113,16 @@ void IDrawingContextListener.OnUpdate(DrawingContext drawingContext, Rect rect)
 
 		void IInputListener.OnInput(InputEvent inputEvent)
 		{
-			switch (inputEvent.Key.Key)
+			if (inputEvent.Key.Key == ScrollUpKey)
 			{
-				case ConsoleKey.UpArrow:
-					Top -= 1;
-					return;
-				case ConsoleKey.DownArrow:
-					Top += 1;
-					break;
-				default:
-					return;
+				Top -= 1;
 			}
-
+			else if (inputEvent.Key.Key == ScrollDownKey)
+			{
+				Top += 1;
+			}
+			
 			inputEvent.Handled = true;
```

Comment:
The `inputEvent.Handled = true;` should only be called if the `inputEvent` was actually handled (so only if a supported key was pressed). This allows for remaining events to be propagated down the chain (for example if you have a text box within a scroll panel and you want to support interactions with both - up/down for scrolling and other keys for input).